src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/mainscreen/viewmodel/MainScreenViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package org.wycliffeassociates.otter.jvm.app.ui.mainscreen.viewmodel
+
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.beans.property.SimpleStringProperty
+import org.wycliffeassociates.otter.common.data.model.ContentLabel
+import org.wycliffeassociates.otter.common.data.workbook.Chapter
+import org.wycliffeassociates.otter.common.data.workbook.Chunk
+import org.wycliffeassociates.otter.common.data.workbook.Workbook
+import org.wycliffeassociates.otter.jvm.app.ui.mainscreen.view.MainScreenView
+import org.wycliffeassociates.otter.jvm.app.ui.cardgrid.view.CardGridFragment
+import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.view.TakeManagementFragment
+import org.wycliffeassociates.otter.jvm.app.ui.workbook.viewmodel.WorkbookViewModel
+import tornadofx.*
+
+class MainScreenViewModel : ViewModel() {
+    private val workbookViewModel: WorkbookViewModel by inject()
+
+    val selectedProjectName = SimpleStringProperty()
+    val selectedProjectLanguage = SimpleStringProperty()
+
+    val selectedChapterTitle = SimpleStringProperty()
+    val selectedChapterBody = SimpleStringProperty()
+
+    val selectedChunkTitle = SimpleStringProperty()
+    val selectedChunkBody = SimpleStringProperty()
+
+    private val takesPageDocked = SimpleBooleanProperty(false)
+
+    init {
+        workbookViewModel.activeWorkbookProperty.onChange {
+            it?.let { wb -> projectSelected(wb) }
+        }
+
+        workbookViewModel.activeChapterProperty.onChange {
+            it?.let { chapter -> chapterSelected(chapter) }
+        }
+
+        workbookViewModel.activeChunkProperty.onChange {
+            it?.let { chunk -> chunkSelected(chunk) }
+                ?: takesPageDocked.set(false)
+        }
+    }
+
+    private fun projectSelected(selectedWorkbook: Workbook) {
+        setActiveProjectText(selectedWorkbook)
+
+        find<MainScreenView>().activeFragment.dock<CardGridFragment>()
+    }
+
+    private fun chapterSelected(chapter: Chapter) {
+        setActiveChapterText(chapter)
+    }
+
+    private fun chunkSelected(chunk: Chunk) {
+        setActiveChunkText(chunk)
+
+        if(takesPageDocked.value == false) {
+            find<MainScreenView>().activeFragment.dock<TakeManagementFragment>()
+        }
+        takesPageDocked.set(true)
```

Comment:
Move into `if`?

Reply:
I wonder if this was just a holdover from some previous version of the code, and if that previous version observed `takesPageDocked` somewhere. (Not sure why else it would be a property.) Anyway, `takesPageDocked` doesn't do anything useful, since the only time the value is read is before deciding whether to dock the `TakeManagementFragment`. But trying to dock a fragment that is already docked just results in a no-op anyway. So I actually just removed `takesPageDocked.`